.github/workflows/crypto3-testing-linux.yml

```
@@ -0,0 +1,209 @@
+
+          if [ "${{ inputs.concurrency }}" = "0" ]; then
+            echo "Setting concurrency to number of logical cores"
+            proc_number=$(nproc)
+          else
+            echo "Setting concurrency to user-defined value"
+            proc_number=${{ inputs.concurrency }}
+          fi
+
+          echo "proc-number=${proc_number}" >> $GITHUB_OUTPUT
+
+      - name: Clean index.lock files if checkout step was cancelled or failed
+        if: cancelled() || failure()
+        run: |
+          find .git -name 'index.lock' -exec rm -v {} \;
+
+      - name: Configure CMake
+        run: >
+          cmake -B build
+          -DCMAKE_CXX_COMPILER=${{ matrix.cpp_compiler }}
+          -DCMAKE_BUILD_TYPE=${{ matrix.build_type }}
+          -DBUILD_TESTS=TRUE
+          -DENABLE_JUNIT_TEST_OUTPUT=TRUE -S ${{ steps.strings.outputs.docker-workspace-path }}
+
+        env:
+          BOOST_ROOT: "${{ steps.install-boost.outputs.BOOST_ROOT }}"
+
+      - name: Build tests
+        id: build
+        working-directory: ${{ steps.strings.outputs.docker-workspace-path }}/build
+        run: |
+          # Not considering failed targets bad. We will handle them as junit test result
+          build_log_path=${{ steps.strings.outputs.docker-workspace-path }}/build/build.log
+          targets_str=$(echo "${{ inputs.targets }}" | awk '{$1=$1};1' | sed '/^$/d' | tr '\n' ' ' | sed 's/ $//')
+          make -k -j ${{ steps.strings.outputs.proc-number }} ${targets_str} 2>&1 | tee ${build_log_path}
+          echo "build-log=$build_log_path" >> "$GITHUB_OUTPUT"
+
+      - name: Generate JUnit Report from build result
+        id: make-build-report
+        uses: NilFoundation/ci-cd/actions/build-log-to-junit@v1
+        with:
+          build-log: ${{ steps.build.outputs.build-log }}
+
+      - name: Run tests
+        id: run_tests
+        working-directory: ${{ steps.strings.outputs.docker-workspace-path }}/build
+        # Not considering failed tests bad. We will compare diff instead
+        run: |
+          artifact_dir=${{ steps.strings.outputs.docker-workspace-path }}/../results_for_uploading_${{github.sha}}
+          mkdir -p $artifact_dir
+          artifact_dir=$(cd ${{ steps.strings.outputs.docker-workspace-path }}/../results_for_uploading_${{github.sha}} && pwd)
+          echo "artifact-dir=$artifact_dir" >> "$GITHUB_OUTPUT"
+
+          custom_tests_dir=$artifact_dir/ubuntu-22.04/${{ matrix.cpp_compiler }}/${{ matrix.build_type }}
+          mkdir -p $custom_tests_dir
+
+          test_paths="${{ inputs.test-paths }}"
+          for dir in $(echo "${test_paths}" | awk 'NF {$1=$1; print}')
+          do
+            cd ${{ steps.strings.outputs.docker-workspace-path }}/build/$dir
+            targets_str=$(find . -type f -executable -printf '%P ' | sed 's/ $//')
+            ctest -v -j ${{ steps.strings.outputs.proc-number }} -R "(${targets_str})" || exit 0
```

```suggestion
            ctest -v -j ${{ steps.strings.outputs.proc-number }} -R "(${targets_str})" || true
```
Now the script exists if test failed